tokio/src/sync/broadcast.rs

```diff
@@ -1213,6 +1330,42 @@ impl<T> Receiver<T> {
 
         Ok(RecvGuard { slot })
     }
+
+    /// Returns the number of [`Sender`] handles.
+    pub fn sender_strong_count(&self) -> usize {
+        self.shared.num_tx.load(SeqCst)
+    }
+
+    /// Returns the number of [`WeakSender`] handles.
+    pub fn sender_weak_count(&self) -> usize {
+        self.shared.num_weak_tx.load(SeqCst)
+    }
+
+    /// Checks if a channel is closed.
+    ///
+    /// This method returns `true` if the channel has been closed. The channel is closed
+    /// when all [`Sender`] have been dropped.
+    ///
+    /// [`Sender`]: crate::sync::broadcast::Sender
+    ///
+    /// # Examples
+    /// ```
+    /// use tokio::sync::broadcast;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let (tx, rx) = broadcast::channel::<()>(10);
+    ///     assert!(!rx.is_closed());
+    ///
+    ///     drop(tx);
+    ///
+    ///     assert!(rx.is_closed());
+    /// }
+    /// ```
+    pub fn is_closed(&self) -> bool {
+        // Channel is closed when there are no strong senders left active
+        self.shared.num_tx.load(Acquire) == 0
```

Comment:
Man, the orderings in this file are a mess. Mixing SeqCst and non-SeqCst orderings on the same atomic is not a good idea. The correct orderings for counters is:

* Increments happen with relaxed.
* Decrements happen with acqrel.
* Checks for zero happen with acquire.

We shouldn't need SeqCst for `num_tx` ever.

Reply:
I'm sorry for that. There orderings and atomic operations are new for me so I copied it from `sync/mpsc` and tried to understand why the specific orderings where used there.
Thank you for some explanation on what should be used instead! I will change the code accordingly.

Reply:
Re-reading your comment I noticed that you said that we don't need `SeqCst` for `num_tx` ever. That was not part of my PR but should I adjust it now here in the PR too?